### PR TITLE
Fix #3167 change ownership of files in build/

### LINKED ISSUE #3167: files created in build/ by docker have root/root ownership
- Version: libbeat on master branch
- Operating System: ALL
- Steps to Reproduce: 

```
cd src/github.com/elastic/beats/libbeat
make testsuite

ll build/
➜  libbeat git:(master) ✗ ll build/     
total 20K
drwxrwxr-x 2 cyrille cyrille 4.0K Dec 10 17:06 coverage
drwxr-xr-x 3 root    root    4.0K Dec 10 17:06 system-tests
-rw-rw-r-- 1 cyrille cyrille   65 Dec 10 17:06 test.env
```

```
cd src/github.com/elastic/beats/winlogbeat
make package

ll build/     
total 48M
drwxrwxr-x 2 cyrille cyrille 4.0K Dec 10 17:02 coverage
drwxr-xr-x 3 root    root    4.0K Dec 10 17:02 homedir
-rwxr-xr-x 1 root    root     11M Dec 10 17:02 import_dashboards-windows-386
-rwxr-xr-x 1 root    root     12M Dec 10 17:02 import_dashboards-windows-amd64
-rw-rw-r-- 1 cyrille cyrille  521 Dec 10 17:02 install-service-winlogbeat.ps1
-rw-rw-r-- 1 cyrille cyrille  279 Dec 10 17:02 package.yml
-rw-rw-r-- 1 cyrille cyrille  188 Dec 10 17:02 uninstall-service-winlogbeat.ps1
drwxrwxr-x 2 cyrille cyrille 4.0K Dec 10 17:03 upload
-rw-r--r-- 1 root    root     23K Dec 10 17:02 winlogbeat-darwin.full.yml
-rw-r--r-- 1 root    root    2.9K Dec 10 17:02 winlogbeat-darwin.yml
-rw-r--r-- 1 root    root     23K Dec 10 17:02 winlogbeat-linux.full.yml
-rw-r--r-- 1 root    root    2.9K Dec 10 17:02 winlogbeat-linux.yml
-rw-r--r-- 1 root    root    5.2K Dec 10 17:02 winlogbeat.template-es2x.json
-rw-r--r-- 1 root    root    4.1K Dec 10 17:02 winlogbeat.template.json
-rwxr-xr-x 1 root    root     12M Dec 10 17:02 winlogbeat-windows-386.exe
-rwxr-xr-x 1 root    root     14M Dec 10 17:02 winlogbeat-windows-amd64.exe
-rw-r--r-- 1 root    root     23K Dec 10 17:02 winlogbeat-win.full.yml
-rw-r--r-- 1 root    root    2.9K Dec 10 17:02 winlogbeat-win.yml
```

### PR DESCRIPTION
**Pull request by cyrilleverrier**
Add a new Makefile rule: fix-permissions

fix-permissions runs a docker container that changes the ownership
of all files from root to the user that runs the Makefile

### CODE REVIEW

**Comment by cyrilleverrier** [comment]
Another solution is to change the user in all docker containers that run the build scripts, using this method:
https://denibertovic.com/posts/handling-permissions-with-docker-volumes/

I started this work but it requires much more changes than this PR.

**Comment by ruflin** [comment]
@cyrilleverrier Thanks for starting the work on this one. It also messes up our cleanup on our jenkins slaves. So more then happy if we find a good solution here. Not sure yet what are the pros and cons of the different options.

**Comment by cyrilleverrier** [comment]
@ruflin I believe this solution is the fastest way to solve the issue with `make clean`.

Other solutions (see my previous comment) requires to change the build scripts and docker images to compile with the current local user instead of `root`. This work gets a bit complicated as the compilation in the docker image should be run in `/home/<user>` folder instead of `/`

**Comment by ruflin** [comment]
@cyrilleverrier You mentioned in the previous comment that it needs lot more work. I assume you commented out the fix-permission line because you hit some issues there? Can you share some details on how we could push this forward?

**Comment by cyrilleverrier** [comment]
@ruflin I just fixed the line in the Makefile and forced a push.

File: libbeat/scripts/Makefile [hunk 1]
```diff
@@ -405,11 +409,15 @@ package: package-setup
...
 
 package-dashboards: package-setup
 	mkdir -p ${BUILD_DIR}
 	cp -r _meta/kibana ${BUILD_DIR}/dashboards
 	# build the dashboards package
 	BEATNAME=${BEATNAME} BUILD_DIR=${BUILD_DIR} SNAPSHOT=$(SNAPSHOT) $(MAKE) -C ${ES_BEATS}/dev-tools/packer package-dashboards ${shell pwd}/build/upload/build_id.txt
+
+fix-permissions:
+	# Change ownership of all files inside /build folder from root/root to current user/group
+	docker run -v ${BUILD_DIR}:/build -it alpine sh -c "chown -R $(shell id -u):$(shell id -g) /build"
```
Comment (ruflin) [inline libbeat/scripts/Makefile]: I would prefer if we use a "fixed" version of alpine here. We had it in the past that an update to a docker image broke the builds.

Comment (cyrilleverrier) [inline libbeat/scripts/Makefile]: Changed to `alpine:3.4`

**Comment by ruflin** [comment]
jenkins, test it

**Comment by cyrilleverrier** [comment]
jenkins, test it

**Comment by ruflin** [comment]
jenkins, test it

**Comment by ruflin** [comment]
@cyrilleverrier Thanks a lot for this one. Finally no manual cleanup anymore on the jenkins slaves hopefully ;-)

**Comment by cyrilleverrier** [comment]
@ruflin thanks for merging. :+1:

**Pull request merged**